streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -1226,7 +1226,11 @@
      */
     public <K, V> KStream<K, V> merge(final KStream<K, V>... streams) {
         try {
```

Comment:
We should add a parameter check at the beginning. Cf. https://github.com/apache/kafka/pull/3916/files#diff-5142e1d4a6410459d6bf6df98828e5afL351